dashboard/app/subsystem.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2023 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/syzkaller/pkg/subsystem"
+	"golang.org/x/net/context"
+	db "google.golang.org/appengine/v2/datastore"
+	"google.golang.org/appengine/v2/log"
+)
+
+// reassignBugSubsystems is expected to be periodically called to refresh old automatic
+// subsystem assignments.
+func reassignBugSubsystems(c context.Context, ns string, count int) error {
+	service := getSubsystemService(c, ns)
+	if service == nil {
+		return nil
+	}
+	keys, err := bugsToUpdateSubsystems(c, ns, count)
+	if err != nil {
+		return err
+	}
+	log.Infof(c, "updating subsystems for %d bugs in %#v", len(keys), ns)
+	now := timeNow(c)
+	for i, bugKey := range keys {
+		if i > 0 {
+			// Prevent contention on db updates.
+			time.Sleep(time.Second)
```

Comment:
Is it really that bad? If yes, I think we are doing something wrong.
What if you move inferSubsystems out of the transaction? I don't think it needs to be in the transaction, and reporting doesn't do it in transaction.

Reply:
I removed the sleep, let's see how it works.